### PR TITLE
feat: add C `ndarray` API and refactor `blas/ext/base/drev`

### PR DESCRIPTION
**Pull request by headlessNode**
## Description

> What is the purpose of this pull request?

This pull request:

-   Adds C `ndarray` API and refactors `blas/ext/base/drev`

## Related Issues

> Does this pull request have any related issues?

No. 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/drev/README.md [hunk 1]
```diff
@@ -141,6 +142,123 @@ console.log( x );
...
+
+### Usage
+
+```c
+#include "stdlib/blas/ext/base/drev.h"
+```
+
+#### stdlib_strided_drev( N, \*X, strideX )
+
+Reverses a double-precision floating-point strided array `x` in-place.
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/drev/README.md]:
```suggestion
Reverses a double-precision floating-point strided array `X` in-place.
```

File: lib/node_modules/@stdlib/blas/ext/base/drev/README.md [hunk 2]
```diff
@@ -141,6 +142,123 @@ console.log( x );
...
+-   **X**: `[in] double*` input array.
+-   **strideX**: `[in] CBLAS_INT` stride length for `X`.
+
+```c
+double stdlib_strided_drev( const CBLAS_INT N, const double *X, const CBLAS_INT strideX );
+```
+
+#### stdlib_strided_drev_ndarray( N, \*X, strideX, offsetX )
+
+Reverses a double-precision floating-point strided array `x` in-place using alternative indexing semantics.
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/drev/README.md]:
```suggestion
Reverses a double-precision floating-point strided array `X` in-place using alternative indexing semantics.
```

**Comment by headlessNode** [comment]
Same `cppcheck` error as other PRs.

File: lib/node_modules/@stdlib/blas/ext/base/drev/README.md [hunk 3]
```diff
@@ -141,6 +142,123 @@ console.log( x );
...
+
+The function accepts the following arguments:
+
+-   **N**: `[in] CBLAS_INT` number of indexed elements.
+-   **X**: `[in] double*` input array.
+-   **strideX**: `[in] CBLAS_INT` stride length for `X`.
+
+```c
+double stdlib_strided_drev( const CBLAS_INT N, const double *X, const CBLAS_INT strideX );
+```
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/drev/README.md]:
```suggestion
```c
double stdlib_strided_drev( const CBLAS_INT N, double *X, const CBLAS_INT strideX );
```
```

File: lib/node_modules/@stdlib/blas/ext/base/drev/README.md [hunk 4]
```diff
@@ -141,6 +142,123 @@ console.log( x );
...
+The function accepts the following arguments:
+
+-   **N**: `[in] CBLAS_INT` number of indexed elements.
+-   **X**: `[in] double*` input array.
+-   **strideX**: `[in] CBLAS_INT` stride length for `X`.
+-   **offsetX**: `[in] CBLAS_INT` starting index for `X`.
+
+```c
+void stdlib_strided_drev_ndarray( const CBLAS_INT N, const double *X, const CBLAS_INT strideX, const CBLAS_INT offsetX );
+```
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/drev/README.md]:
```suggestion
```c
void stdlib_strided_drev_ndarray( const CBLAS_INT N, double *X, const CBLAS_INT strideX, const CBLAS_INT offsetX );
```
```

File: lib/node_modules/@stdlib/blas/ext/base/drev/README.md [hunk 5]
```diff
@@ -47,9 +47,9 @@ The function has the following parameters:
 
 -   **N**: number of indexed elements.
 -   **x**: input [`Float64Array`][@stdlib/array/float64].
--   **stride**: index increment.
+-   **strideX**: stride length.
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/drev/README.md]:
```suggestion
-   **strideX**: stride length for `x`.
```

File: lib/node_modules/@stdlib/blas/ext/base/drev/README.md [hunk 6]
```diff
@@ -91,9 +91,9 @@ drev.ndarray( x.length, x, 1, 0 );
 
 The function has the following additional parameters:
 
--   **offset**: starting index.
+-   **offsetX**: starting index.
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/drev/README.md]:
```suggestion
-   **offsetX**: starting index for `x`.
```

**Review by Planeshifter — APPROVED** [review]
Not spotting any issues. Let's get this in, thanks @headlessNode!

**Pull request merged**